compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -208,16 +210,24 @@ impl<'tcx> fmt::Display for LayoutError<'tcx> {
             LayoutError::SizeOverflow(ty) => {
                 write!(f, "values of the type `{}` are too big for the current architecture", ty)
             }
+            LayoutError::NormalizationFailure(t, e) => write!(
+                f,
+                "unable to determine layout for `{}` because `{}` cannot be normalized",
```

Comment:
So, this is never actually emitted? Makes me wonder if we actually need to keep track of what we couldn't normalize.

Reply:
yeah I noticed that too and was debating whether we need the type, but think that it makes sense to output the type for which normalization fails here. There are a couple other issues which triggered this ICE (but not through `layout_of` and which have yet to be fixed (although I think the new function would make that fairly straightforward)), and we might need this type information there. 

I also added another test for issue 85103 which emits this error.